src/hooks/useConsent.ts

```diff
@@ -13,9 +17,18 @@ export const useConsent = () => {
     });
   };
 
-  const useUpdateProjectConsentDirective = () => {
+  const useUpdateProjectConsentDirective = (
+    options: UseMutationOptions<
+      PatchConsentDirectives['Response'],
+      unknown,
+      PatchConsentDirectives['Request'] & {
+        directiveId: string;
+      }
+    > = {},
```

Comment:
Rather than provide the full options here, can we _just_ provide an `onSuccess` option? That will simplify the typing a lot, and make a soon-to-come change from me a lot easier.